contracts/types/LockData.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.20;
+
+/// @notice The LockData holds global information for the lock data structure: the length and nonzeroDeltaCount.
+/// @dev The left most 128 bits is the length, or total number of active lockers.
+/// @dev The right most 128 bits is the nonzeroDeltaCount, or total number of nonzero deltas over all active + completed lockers.
+/// @dev Located in transient storage.
+type LockData is uint256;
+
+using LockDataLibrary for LockData global;
+
+function toLockData(uint128 _length, uint128 _nonzeroDeltaCount) pure returns (LockData lockData) {
+    /// @solidity memory-safe-assembly
+    assembly {
+        lockData :=
+            or(
+                shl(128, _length),
+                and(0x00000000000000000000000000000000ffffffffffffffffffffffffffffffff, _nonzeroDeltaCount)
+            )
+    }
+}
+
+/// @dev This library manages a custom storage implementation for a queue
+///      that tracks current lockers. The LOCK_DATA storage slot for this data structure,
+///      passed in as a LockData custom type that is a uint256, stores not just the current
+///      length of the queue but also the global count of non-zero deltas across all lockers.
+///      The values of the data structure start at LOCKERS, and each value is a locker address.
+library LockDataLibrary {
```

Comment:
I found this library quite hard to read tbh. i think a few things that would help:
- Function ordering - currently you have to really jump around to read the library. e.g. `_update` is defined in the middle of the library, and there are functions both above and below it that call it. I'd personally rather the functions were in an order where all the function they call are defined above it (or below it if you prefer).
- Comments! Lots of the functions have no comments at all